libvips/include/vips/foreign.h

```diff
@@ -330,13 +330,48 @@ typedef enum {
 	VIPS_SAVEABLE_LAST
 } VipsSaveable;
 
+/**
+ * VipsForeignPreserve:
+ * @VIPS_FOREIGN_PRESERVE_NONE: don't attach metadata
+ * @VIPS_FOREIGN_PRESERVE_EXIF: preserve Exif metadata
+ * @VIPS_FOREIGN_PRESERVE_XMP: preserve XMP metadata
+ * @VIPS_FOREIGN_PRESERVE_IPTC: preserve IPTC metadata
+ * @VIPS_FOREIGN_PRESERVE_ICC: preserve ICC metadata
+ * @VIPS_FOREIGN_PRESERVE_OTHER: preserve other metadata (e.g. PNG comments and some TIFF tags)
+ * @VIPS_FOREIGN_PRESERVE_ALL: preserve all metadata
+ *
+ * Which metadata should be preserved.
+ */
+typedef enum /*< flags >*/ {
+	VIPS_FOREIGN_PRESERVE_NONE = 0,
+	VIPS_FOREIGN_PRESERVE_EXIF = 1 << 0,
+	VIPS_FOREIGN_PRESERVE_XMP = 1 << 1,
+	VIPS_FOREIGN_PRESERVE_IPTC = 1 << 2,
+	VIPS_FOREIGN_PRESERVE_ICC = 1 << 3,
+	VIPS_FOREIGN_PRESERVE_OTHER = 1 << 4,
+
+	VIPS_FOREIGN_PRESERVE_ALL = (VIPS_FOREIGN_PRESERVE_EXIF |
```

Comment:
Could we just use `-1`?

Reply:
I think flags values should always be a signed integer, as `vips_flags_from_nick` uses `-1` to indicate an error.
https://github.com/libvips/libvips/blob/7513fa3bf5ffaf388da57fc42becf94f7c4e6b64/libvips/iofuncs/object.c#L2120-L2121
https://github.com/libvips/pyvips/blob/a2b08d4c5aa203399d03b379ca05145f32ce5b94/pyvips/gvalue.py#L136-L138

Reply:
`g_value_set_flags` also expects flags to be unsigned.
https://github.com/GNOME/glib/blob/2.78.0/gobject/genums.c#L741